src/components/connect-request.js

```diff
@@ -27,18 +29,33 @@ const connectRequest = (mapPropsToConfig, options = {}) => (WrappedComponent) =>
         }
 
         componentDidMount() {
-            this.requestAsync(this.props, false, true);
+            const configs = mapPropsToConfig(this.props);
+            if (Array.isArray(configs)) {
+                this.requestAsync(configs, false, true);
+            } else {
+                this.requestAsync(this.props, false, true);
+            }
         }
 
         componentDidUpdate(prevProps) {
-            const prevConfig = mapPropsToConfig(prevProps);
-            const config = mapPropsToConfig(this.props);
-            const prevQueryKey = prevConfig && getQueryKey(prevConfig.url, prevConfig.body);
-            const queryKey = config && getQueryKey(config.url, config.body);
+            let prevConfigs = mapPropsToConfig(prevProps);
+            let configs = mapPropsToConfig(this.props);
+            if (!Array.isArray(prevConfigs)) {
+                prevConfigs = [prevConfigs];
+            }
+            if (!Array.isArray(configs)) {
+                configs = [configs];
+            }
 
-            if (prevQueryKey !== queryKey) {
-                this.cancelPendingRequests();
-                this.requestAsync(this.props, false, true);
+            const diffs = this._diffConfigs(prevConfigs, configs);
+            const toCancel = diffs[0];
```

Comment:
Can you rename the variables here to indicate their types. cancel == keys and requests are configs. Alternatively, maybe makes sense to always return keys from `diffConfigs` and do the mapping from query keys to configs here in `componentDidUpdate`.